packages/e2e-tests/specs/performance/performance.test.js

```diff
@@ -49,35 +55,62 @@ describe( 'Performance', () => {
 		};
 
 		let i = 1;
-		let startTime;
-
-		await page.on( 'load', () =>
-			results.load.push( new Date() - startTime )
-		);
-		await page.on( 'domcontentloaded', () =>
-			results.domcontentloaded.push( new Date() - startTime )
-		);
 
 		while ( i-- ) {
-			startTime = new Date();
 			await page.reload( { waitUntil: [ 'domcontentloaded', 'load' ] } );
+			const timings = JSON.parse(
+				await page.evaluate( () =>
+					JSON.stringify( window.performance.timing )
+				)
+			);
+			const {
+				navigationStart,
+				domContentLoadedEventEnd,
+				loadEventEnd,
+			} = timings;
+			results.load.push( loadEventEnd - navigationStart );
+			results.domcontentloaded.push(
+				domContentLoadedEventEnd - navigationStart
+			);
 		}
 
 		await insertBlock( 'Paragraph' );
 
 		i = 200;
+		const traceFile = __dirname + '/trace.json';
+
+		await page.tracing.start( {
+			path: traceFile,
+			screenshots: false,
+			categories: [ 'devtools.timeline' ],
+		} );
 
 		while ( i-- ) {
-			startTime = new Date();
 			await page.keyboard.type( 'x' );
-			results.type.push( new Date() - startTime );
 		}
 
+		await page.tracing.stop();
+
+		const traceResults = JSON.parse( readFile( traceFile ) );
+		const isKeypressEvent = ( item ) =>
+			item.cat === 'devtools.timeline' &&
+			item.name === 'EventDispatch' &&
+			item.dur &&
+			item.args &&
+			item.args.data &&
+			item.args.data.type === 'keypress';
```

Comment:
I'm not sure that's enough, I know previously keydown was also a problem.

<img width="520" alt="Capture d’écran 2020-03-12 à 11 16 51 AM" src="https://user-images.githubusercontent.com/272444/76511182-1411de80-6453-11ea-9d47-7a8cbfe9e6d4.png">

You can see in the screenshot that there's aa keydown and then a keypress. Both of these are gathered in a "Task". and after that there are "follow up async tasks).

Ideally, if we can meansure the three segments it would be great, I believe the most important thing is the sync behavior (keydown + keypress). A format like that might be best `keydown+keypress ( follow-ups: xx )` eg: `25ms ( follow-ups: 40ms )` 

What do you think?

Reply:
I don't think it will make much difference, since I expect the absolute values not to be as important as their evolution over time, and the `keydown`/`keyup` events don't seem to have a great deal of variability.

That said, for completeness sake, I added a commit where I calculate the sum of the three event durations for a total value for each virtual key press.

Reply:
I just noticed that you'd suggested separating out the value in the output. Do you think that's necessary, or is the sum enough?

Reply:
That sum is enough for the first part (the synchronous work). I don't know though if it's possible somehow to compute the time spent in the asynchronous work. In the trace it's shown as separate from keydown, keyup, keypress event, it's shown as multiple small tasks.

I'd say, getting that value is less important but if we manage to get it, we shouldn't sum it with the rest.

does that make sense?

Reply:
I'd argue that whether or not the async value is important depends on what you're trying to measure. If you're trying to measure input latency as perceived by the user, then the async value isn't important, as that work happens after the screen has updated and probably does not limit interactivity in any way, as the tasks are very short.

In any case, I suspect it would be extremely difficult to get that value, as there wouldn't be a good way of correlating the async work with the key events that triggered it.